modeling/dataset/samplers/base.py

```diff
@@ -104,16 +106,13 @@ def __getitem__(self, index):
 
 class MultiDomainEvalSampler(EvalSampler):
 
-    def __init__(self, dataset, num_users, num_items, target_domain):
+    def __init__(self, dataset, num_users, num_items):
         super().__init__()
         self._dataset = dataset
         self._num_users = num_users
         self._num_items = num_items
-        self._target_domain = target_domain
-
-    @property
-    def dataset(self):
-        return self._dataset
+        self._target_domain = None
+        self._other_domains = None
 
     def __len__(self, domain):
```

Comment:
I checked `__len__` definition required for pytorch DataLoader. Let's re-write it in the following way: 

```
def __len__(self, domain = None):
  if domain is None:
    return len(self._dataset[self._target_domain])
  return len(self._dataset[domain])
```